projects/igniteui-angular-elements/src/lib/state.component.ts

```diff
@@ -1,7 +1,34 @@
 import { Input, Component, EnvironmentInjector, Inject, Injector, ViewContainerRef } from '@angular/core';
-import {IgxGridStateDirective, IGridStateOptions, IGridState, GridFeatures } from "../../../igniteui-angular/src/lib/grids/state.directive";
-import { IGX_GRID_SERVICE_BASE, GridServiceType}  from '../../../igniteui-angular/src/lib/grids/common/grid.interface';
+import {IgxGridStateDirective, IGridStateOptions, IGridState, GridFeatures, IColumnState, IGridStateCollection } from "../../../igniteui-angular/src/lib/grids/state.directive";
+import { IGX_GRID_SERVICE_BASE, GridServiceType, IPinningConfig}  from '../../../igniteui-angular/src/lib/grids/common/grid.interface';
 import { IgxGridComponent } from '../../../igniteui-angular/src/lib/grids/grid/grid.component';
+import { IFilteringExpressionsTree } from '../../../igniteui-angular/src/lib/data-operations/filtering-expressions-tree';
+import { IPagingState } from '../../../igniteui-angular/src/lib/data-operations/paging-state.interface';
+import { ISortingExpression } from '../../../igniteui-angular/src/lib/data-operations/sorting-strategy';
+import { IGroupingState } from '../../../igniteui-angular/src/lib/data-operations/groupby-state.interface';
+import { GridSelectionRange } from '../../../igniteui-angular/src/lib/grids/common/types';
+import { IPivotConfiguration } from '../../../igniteui-angular/src/lib/grids/pivot-grid/pivot-grid.interface'
+
+/* tsPlainInterface */
+/* marshalByValue */
+export interface IState {
```

Comment:
If we have no other options to avoid the naming conflict, feel like `State` might be a bit too generic of a name - both not descriptive enough and also likely easy to conflict with other unrelated types and should likely still contain `Grid` in the name. So if we can't have `GridState` for this as well, perhaps add words rather than remove - think `GridStateResult` or `GridStateSnapshot`. Probably a few good alternatives out there.